File: evenfibonaccinumber/python/fransay.py
# Author: fransay
# Date: 01/05/2023
# Description: Project Euler (Sum of even fibonacci numbers)

import time as t


def even(number) -> bool:
    """
    returns a boolean representation of the even status
    :param number:
    :return:
    """
    is_even = number % 2 == 0
    return is_even


def main():
    start_time = t.time()  # init stop clock
    LIMIT = 4000000
    SUM = 0
    count_number = 0
    start_count_number, next_count_number = count_number, count_number + 1
    while start_count_number <= LIMIT:
        third_number = start_count_number + next_count_number
        if even(third_number):
            SUM = SUM + third_number
        start_count_number, next_count_number = next_count_number, third_number
        count_number += 1
        # print(third_number, end=" \n")
    print("SUM: ", SUM)
    end_time = t.time()  # end stop clock counter in seconds
    time_elapsed = end_time - start_time
    print(f'Runtime: {time_elapsed} seconds')


if __name__ == '__main__':
    main()

File: evenfibonaccinumber/dart/fransay.dart
// Author: Fransay
// Date: 29/04/2023
// Description: Project Euler(Sum of Even Fibonaccinumbers)
// Complexity: O(n)


void main(){
  int countNumber = 0; // counting number starting at zero
  var sumEvenFib = 0; // sum of even fibonacci numbers
  const LIMIT = 4000000; // fibonacci sequence must not exceed LIMIT
  var firstFibNumber = countNumber; // first fib number 
  var secondFibNumber = firstFibNumber + 1; // second fib number (firstNumber + 1)

  while (firstFibNumber <= LIMIT){
    var thirdFibNumber = firstFibNumber + secondFibNumber;
    if (even(thirdFibNumber)){
      sumEvenFib = sumEvenFib + thirdFibNumber;
    }
    firstFibNumber = secondFibNumber;
    secondFibNumber = thirdFibNumber;
  }
  print("SUM OF EVEN FIB NUMBERS: $sumEvenFib"); 
}

// even function: checks if a number is even or not
bool even(int number){
  if (number % 2 == 0){
    return true;
  }
  return false;
}